store/write_controlled_store.go

```diff
@@ -18,13 +19,18 @@ type WriteControlledStore struct {
 
 	lock       sync.Mutex
 	entryTable map[imap.InternalMessageID]*syncRef
-	lockPool   []*syncRef
+	lockPool   sync.Pool
 }
 
 func NewWriteControlledStore(impl Store) *WriteControlledStore {
 	return &WriteControlledStore{
 		impl:       impl,
 		entryTable: make(map[imap.InternalMessageID]*syncRef),
+		lockPool: sync.Pool{
+			New: func() interface{} {
```

Comment:
```suggestion
			New: func() any {
```

with go 1.18 `interface{}` is pretty much deprecated, use `any` wherever you can

Reply:
Done